src/types/message.ts

```diff
@@ -103,6 +79,28 @@ export interface StopMessage extends MessageBase {
   type: 'stop';
 }
 
+export interface GeofenceMessage extends MessageBase {
+  type: 'geofence';
+
+  /**
+    * Geofencing coordinates in the form of a rectangle.
+    */
+  geofence: {
+    'keepOut': [number, number][],
+    'keepIn': [number, number][]
+  }
+}
+
+/**
+ * Type guard for Geofence Message.
+ */
+function isGeofenceMessage(message: Message): boolean {
+  return message.type === 'geofence'
+    && message.geofence !== undefined
+    && Array.isArray(message.geofence.keepIn)
```

Comment:
lests add a check for each array to verify that each member of the array is [number, number]:
```ts
mission.geofence.keepIn.every((coords) => coords.length === 2 && Number.isNumber(coords[0]) && Number.isNumber(coords[1]));
```